Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,785 @@
+        last_run (dateparser.time): The greatest incident created_time we fetched from last fetch
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Demisto
+    """
+    time_format = '%Y-%m-%dT%H:%M:%SZ'
+    if not last_run:  # if first time running
+        new_last_run = {'last_fetch': parse_date_range(fetch_time, date_format=time_format)[0]}
+    else:
+        new_last_run = last_run
+
+    parsed_last_time = datetime.strptime(new_last_run.get('last_fetch', ''), time_format)
+    filter_after_last_run = f'created gt {parsed_last_time}'
+    fetch_filter_for_query = f'{filter_after_last_run}'
+    if fetch_queue_id:
+        filter_by_queue_id = f'hd_queue_id eq {fetch_queue_id}'
+        fetch_filter_for_query = f'{fetch_filter_for_query},{filter_by_queue_id}'
+    if fetch_filter:
+        fetch_filter_for_query = f'{fetch_filter_for_query},{fetch_filter}'
+    if not fetch_shaping:
+        fetch_shaping = set_shaping(client, fetch_queue_id)
+    demisto.info(f"Fetching Incident has Started,\n"
+                 f"Fetch filter is {fetch_filter_for_query}\n"
+                 f"Last fetch was on {str(parsed_last_time)}")
+    client.update_token()
+    items: dict = client.tickets_list_request(fetch_shaping, fetch_filter_for_query)
+    items: list = items.get('Tickets', [])
+    incidents, parsed_last_time = parse_incidents(items, fetch_limit, time_format, parsed_last_time)
+    parsed_last_time = parsed_last_time.strftime(time_format)
+    demisto.info(f"Fetching Incident has Finished\n"
+                 f"Fetch limit was {fetch_limit}"
+                 f"Last fetch was on {str(parsed_last_time)}\n"
+                 f"Incidents list is {str(incidents)}")
+    demisto.setLastRun({'last_fetch': parsed_last_time})
+    return incidents
+
+
+def get_fields_by_queue(client, queue: Optional[str]) -> list:
+    if queue:
+        queues_id = [queue]
+    else:
+        queues = client.queues_list_request()
+        queues = queues.get('Queues', [])
+        queues_id = []
+        for q in queues:
+            queues_id.append(q.get('id'))
+    fields: list = []
+    for q in queues_id:
+        client.update_token()
+        fields_by_queue = client.queues_list_fields_request(queue_number=str(q))
+        fields_by_queue = fields_by_queue.get('Fields', [])
+        for field in fields_by_queue:
+            if field.get('jsonKey') not in fields:
+                # get internal error 500 from server with related tickets
+                if field.get('jsonKey') != 'related_tickets' and field.get('jsonKey') != 'referring_tickets':
+                    fields.append(field.get('jsonKey'))
+    return fields
+
+
+def shaping_by_fields(fields: list) -> str:
+    shaping = 'hd_ticket all'
+    for field in fields:
+        shaping = f'{shaping},{field} limited'
```

Comment:
```suggestion
        shaping+ = f',{field} limited'
```
Should not it concat the fields?